blueflood-http/src/main/java/com/rackspacecloud/blueflood/tracker/Tracker.java

```diff
@@ -40,12 +40,27 @@
     private static final String trackerName = String.format("com.rackspacecloud.blueflood.tracker:type=%s", Tracker.class.getSimpleName());
     private static final Pattern patternGetTid = Pattern.compile( "/v\\d+\\.\\d+/([^/]+)/.*" );
 
+    // Tracker is a singleton
+    private static Tracker instance = new Tracker();
+    private static boolean isRegistered = false;
+
     static Set tenantIds = new HashSet();
     static boolean isTrackingDelayedMetrics = false;
     static Set<String> metricNames = new HashSet<String>();
 
-    public Tracker() {
-        registerMBean();
+    // private constructor for singleton
+    private Tracker(){}
+
+    public static Tracker getInstance() {
+        register();
+        return instance;
+    }
+
+    public static void register() {
+        if (!isRegistered) {
```

Comment:
I'll add it but I don't think this will ever happen as you described.  Tracker is not ran in multitude threads (only registered by two servers service on startup), and the register() method only register the MBean, not instantiate the object.  The object is only instantiate once on class loading.